core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -5346,29 +5326,55 @@ public void setKeyRotationTime(@Nullable Date time) {
      * The given time cannot be in the future.
      * </p>
      */
-    public void setKeyRotationTime(long unixTimeSeconds) {
-        checkArgument(unixTimeSeconds <= TimeUtils.currentTimeSeconds(), "Given time (%s) cannot be in the future.",
-                TimeUtils.dateTimeFormat(unixTimeSeconds * 1000));
-        vKeyRotationTimestamp = unixTimeSeconds;
+    public void setKeyRotationTime(Instant keyRotationTime) {
+        checkArgument(keyRotationTime.compareTo(TimeUtils.currentTime()) <= 0,
+                "Given time (%s) cannot be in the future.",
+                TimeUtils.dateTimeFormat(keyRotationTime.toEpochMilli()));
+        vKeyRotationTime = keyRotationTime;
         saveNow();
     }
 
+    /** @deprecated use {@link #setKeyRotationTime(Instant)} */
+    @Deprecated
+    public void setKeyRotationTime(long timeSecs) {
+        if (timeSecs == 0)
+            vKeyRotationTime = null;
+        else
+            setKeyRotationTime(Instant.ofEpochSecond(timeSecs));
+    }
+
+    /** @deprecated use {@link #setKeyRotationTime(Instant)} */
+    @Deprecated
+    public void setKeyRotationTime(@Nullable Date time) {
+        if (time == null)
+            vKeyRotationTime = null;
+        else
+            setKeyRotationTime(Instant.ofEpochMilli(time.getTime()));
+    }
+
     /**
-     * Returns the key rotation time, or null if unconfigured. See {@link #setKeyRotationTime(Date)} for a description
+     * Returns the key rotation time, or empty if unconfigured. See {@link #setKeyRotationTime(Instant)} for a description
```

Comment:
This is one of the few places where the `null`/`empty` case is documented. I think this could be improved.

Reply:
I added another comment to the  `vKeyRotationTime` field.